src/commands/generate.ts

```diff
@@ -52,6 +52,54 @@ async function generate(toolbox: GluegunToolbox) {
     pascalName = pascalName.slice(0, -1 * pascalGenerator.length)
     command(`npx ignite-cli generate ${generator} ${pascalName}`)
   }
+  // Check if src/app exists, denoting an Expo Router app
+  if (generator === "screen") {
+    const packageJson = filesystem.read("package.json", "json")
+    const isExpoRouterApp = !!packageJson?.dependencies?.["expo-router"]
+
+    if (isExpoRouterApp) {
+      const directoryDirSetInFrontMatter = frontMatterDirectoryDir("screen")
+
+      if (directoryDirSetInFrontMatter || dir) {
+        heading(
+          `It looks like you're working in a project using Expo Router, determined directory for screen from ${dir ? "override" : "template front matter"}`,
+        )
+        dir = dir || directoryDirSetInFrontMatter
+      } else {
+        const result = await prompt.ask({
+          type: "input",
+          name: "dir",
+          message:
+            "It looks like you're working in a project using Expo Router, please enter the desired directory (e.g., src/app):",
+        })
+
+        if (result.dir) {
+          // Validate the directory
+          const isValidDir = filesystem.exists(result.dir) === "dir"
+          if (isValidDir) {
+            dir = result.dir
+          } else {
+            const createDirResult = await prompt.ask({
+              type: "confirm",
+              name: "createDir",
+              message: `⚠️  Directory ${result.dir} does not exist. Would you like to create it?`,
+              initial: true,
+              format: prettyPrompt.format.boolean,
+            })
+
+            if (createDirResult.createDir) {
+              filesystem.dir(result.dir)
+              dir = result.dir
+            } else {
+              warning(`⚠️ Placing screen in src/app root.`)
+              p()
+              dir = "src/app"
```

Comment:
I'm thinking this still needs to detect if `src/app` exists or `app` exists - as it could be either if not coming from an Igntie project

maybe we can set it to be the default of the prompt since it's the example in the message anyway